packages/e2e-tests/specs/experiments/blocks/navigation.test.js

```diff
@@ -343,12 +343,14 @@ describe( 'Navigation', () => {
 
 			await selectDropDownOption( 'Test Menu 2' );
 
-			await page.waitForSelector( '.wp-block-navigation__container' );
+			await page.waitForSelector(
+				'.interface-interface-skeleton__content .wp-block-navigation__container'
```

Comment:
I think this line should wait for a navigation link element instead.

The cause of the flakiness might be that the container element is briefly visible before the inner link blocks are added, and the test sometimes makes its assertion a little too early. The code seems to show this might happen. One is a state update, the other is a store update, so they might be batched as separate DOM updates:
https://github.com/WordPress/gutenberg/blob/4f72d04df2c1e994fe391df3cd4c0f5f927a3037/packages/block-library/src/navigation/edit.js#L205-L211

Reply:
That's a good idea, looks like CI is also happy with this in https://github.com/WordPress/gutenberg/pull/33704/commits/f6c25fe62232c02aafd9e31bd2503aaa7fbad74b